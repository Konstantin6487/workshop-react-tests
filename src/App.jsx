import React from 'react';

import 'react-tabs/style/react-tabs.css';

import BasicComponent from './BasicComponent';
import CustomTabComponent from './CustomTabComponent';

const App = () => (
  <div>
    <BasicComponent />
    <CustomTabComponent />
  </div>
);

export default App;
