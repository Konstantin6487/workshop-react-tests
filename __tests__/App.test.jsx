import React from 'react';
import App from '../src/App';

test('App full render', () => {
  const wrapper = mount(<App />);
  expect(wrapper.render()).toMatchSnapshot();
});

test('Tabs container 1', () => {
  const wrapper = mount(<App />);
  expect(wrapper.render()).toMatchSnapshot();

  const tabsContainers = wrapper.find('[data-test="tabs-container"]');
  const firstContainer = tabsContainers.at(1);
  const tabsAnchors = firstContainer.find('[data-test="tab-anchor"]');

  const leftTab = tabsAnchors.at(1);
  const rightTab = tabsAnchors.at(2);

  leftTab.simulate('click');
  expect(leftTab.render()).toMatchSnapshot();

  rightTab.simulate('click');
  expect(rightTab.render()).toMatchSnapshot();
});

test('Tabs container 2', () => {
  const wrapper = mount(<App />);
  expect(wrapper.render()).toMatchSnapshot();

  wrapper.html();
  const tabsContainers = wrapper.find('[data-test="tabs-container"]');
  const secondContainer = tabsContainers.at(2);
  const tabsAnchors = secondContainer.find('[data-test="tab-anchor"]');

  const leftTab = tabsAnchors.at(1);
  const rightTab = tabsAnchors.at(2);

  leftTab.simulate('click');
  expect(leftTab.render()).toMatchSnapshot();

  rightTab.simulate('click');
  expect(rightTab.render()).toMatchSnapshot();
  // eslint-disable-next-line
  debugger;
});
