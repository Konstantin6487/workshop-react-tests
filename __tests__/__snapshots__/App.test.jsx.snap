// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App full render 1`] = `
<div>
  <div
    class="react-tabs"
    data-tabs="true"
  >
    <ul
      class="react-tabs__tab-list"
      data-test="tabs-container"
      role="tablist"
    >
      <li
        aria-controls="react-tabs-1"
        aria-disabled="false"
        aria-selected="true"
        class="react-tabs__tab react-tabs__tab--selected"
        data-test="tab-anchor"
        id="react-tabs-0"
        role="tab"
        tabindex="0"
      >
        Title 1
      </li>
      <li
        aria-controls="react-tabs-3"
        aria-disabled="false"
        aria-selected="false"
        class="react-tabs__tab"
        data-test="tab-anchor"
        id="react-tabs-2"
        role="tab"
      >
        Title 2
      </li>
    </ul>
    <div
      aria-labelledby="react-tabs-0"
      class="react-tabs__tab-panel react-tabs__tab-panel--selected"
      id="react-tabs-1"
      role="tabpanel"
    >
      <h2>
        Any content 1
      </h2>
    </div>
    <div
      aria-labelledby="react-tabs-2"
      class="react-tabs__tab-panel"
      id="react-tabs-3"
      role="tabpanel"
    />
  </div>
  <div
    class="react-tabs"
    data-tabs="true"
  >
    <ul
      class="react-tabs__tab-list"
      data-test="tabs-container"
      role="tablist"
    >
      <li
        aria-disabled="false"
        aria-selected="false"
        class="react-tabs__tab"
        data-test="tab-anchor"
        role="tab"
      >
        <h1>
          Custom Tab 1
        </h1>
      </li>
      <li
        aria-disabled="false"
        aria-selected="false"
        class="react-tabs__tab"
        data-test="tab-anchor"
        role="tab"
      >
        <h1>
          Custom Tab 2
        </h1>
      </li>
    </ul>
    <div
      aria-labelledby="react-tabs-4"
      class="react-tabs__tab-panel react-tabs__tab-panel--selected"
      id="react-tabs-5"
      role="tabpanel"
    >
      Panel 1
    </div>
    <div
      aria-labelledby="react-tabs-6"
      class="react-tabs__tab-panel"
      id="react-tabs-7"
      role="tabpanel"
    />
  </div>
</div>
`;

exports[`Tabs container 1 1`] = `
<div>
  <div
    class="react-tabs"
    data-tabs="true"
  >
    <ul
      class="react-tabs__tab-list"
      data-test="tabs-container"
      role="tablist"
    >
      <li
        aria-controls="react-tabs-9"
        aria-disabled="false"
        aria-selected="true"
        class="react-tabs__tab react-tabs__tab--selected"
        data-test="tab-anchor"
        id="react-tabs-8"
        role="tab"
        tabindex="0"
      >
        Title 1
      </li>
      <li
        aria-controls="react-tabs-11"
        aria-disabled="false"
        aria-selected="false"
        class="react-tabs__tab"
        data-test="tab-anchor"
        id="react-tabs-10"
        role="tab"
      >
        Title 2
      </li>
    </ul>
    <div
      aria-labelledby="react-tabs-8"
      class="react-tabs__tab-panel react-tabs__tab-panel--selected"
      id="react-tabs-9"
      role="tabpanel"
    >
      <h2>
        Any content 1
      </h2>
    </div>
    <div
      aria-labelledby="react-tabs-10"
      class="react-tabs__tab-panel"
      id="react-tabs-11"
      role="tabpanel"
    />
  </div>
  <div
    class="react-tabs"
    data-tabs="true"
  >
    <ul
      class="react-tabs__tab-list"
      data-test="tabs-container"
      role="tablist"
    >
      <li
        aria-disabled="false"
        aria-selected="false"
        class="react-tabs__tab"
        data-test="tab-anchor"
        role="tab"
      >
        <h1>
          Custom Tab 1
        </h1>
      </li>
      <li
        aria-disabled="false"
        aria-selected="false"
        class="react-tabs__tab"
        data-test="tab-anchor"
        role="tab"
      >
        <h1>
          Custom Tab 2
        </h1>
      </li>
    </ul>
    <div
      aria-labelledby="react-tabs-12"
      class="react-tabs__tab-panel react-tabs__tab-panel--selected"
      id="react-tabs-13"
      role="tabpanel"
    >
      Panel 1
    </div>
    <div
      aria-labelledby="react-tabs-14"
      class="react-tabs__tab-panel"
      id="react-tabs-15"
      role="tabpanel"
    />
  </div>
</div>
`;

exports[`Tabs container 1 2`] = `
<li
  aria-controls="react-tabs-9"
  aria-disabled="false"
  aria-selected="true"
  class="react-tabs__tab react-tabs__tab--selected"
  data-test="tab-anchor"
  id="react-tabs-8"
  role="tab"
  tabindex="0"
>
  Title 1
</li>
`;

exports[`Tabs container 1 3`] = `
<li
  aria-controls="react-tabs-11"
  aria-disabled="false"
  aria-selected="true"
  class="react-tabs__tab react-tabs__tab--selected"
  data-test="tab-anchor"
  id="react-tabs-10"
  role="tab"
  tabindex="0"
>
  Title 2
</li>
`;

exports[`Tabs container 2 1`] = `
<div>
  <div
    class="react-tabs"
    data-tabs="true"
  >
    <ul
      class="react-tabs__tab-list"
      data-test="tabs-container"
      role="tablist"
    >
      <li
        aria-controls="react-tabs-17"
        aria-disabled="false"
        aria-selected="true"
        class="react-tabs__tab react-tabs__tab--selected"
        data-test="tab-anchor"
        id="react-tabs-16"
        role="tab"
        tabindex="0"
      >
        Title 1
      </li>
      <li
        aria-controls="react-tabs-19"
        aria-disabled="false"
        aria-selected="false"
        class="react-tabs__tab"
        data-test="tab-anchor"
        id="react-tabs-18"
        role="tab"
      >
        Title 2
      </li>
    </ul>
    <div
      aria-labelledby="react-tabs-16"
      class="react-tabs__tab-panel react-tabs__tab-panel--selected"
      id="react-tabs-17"
      role="tabpanel"
    >
      <h2>
        Any content 1
      </h2>
    </div>
    <div
      aria-labelledby="react-tabs-18"
      class="react-tabs__tab-panel"
      id="react-tabs-19"
      role="tabpanel"
    />
  </div>
  <div
    class="react-tabs"
    data-tabs="true"
  >
    <ul
      class="react-tabs__tab-list"
      data-test="tabs-container"
      role="tablist"
    >
      <li
        aria-disabled="false"
        aria-selected="false"
        class="react-tabs__tab"
        data-test="tab-anchor"
        role="tab"
      >
        <h1>
          Custom Tab 1
        </h1>
      </li>
      <li
        aria-disabled="false"
        aria-selected="false"
        class="react-tabs__tab"
        data-test="tab-anchor"
        role="tab"
      >
        <h1>
          Custom Tab 2
        </h1>
      </li>
    </ul>
    <div
      aria-labelledby="react-tabs-20"
      class="react-tabs__tab-panel react-tabs__tab-panel--selected"
      id="react-tabs-21"
      role="tabpanel"
    >
      Panel 1
    </div>
    <div
      aria-labelledby="react-tabs-22"
      class="react-tabs__tab-panel"
      id="react-tabs-23"
      role="tabpanel"
    />
  </div>
</div>
`;

exports[`Tabs container 2 2`] = `
<li
  aria-disabled="false"
  aria-selected="false"
  class="react-tabs__tab"
  data-test="tab-anchor"
  role="tab"
>
  <h1>
    Custom Tab 1
  </h1>
</li>
`;

exports[`Tabs container 2 3`] = `
<li
  aria-disabled="false"
  aria-selected="false"
  class="react-tabs__tab"
  data-test="tab-anchor"
  role="tab"
>
  <h1>
    Custom Tab 2
  </h1>
</li>
`;
